Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -316,11 +316,20 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
 
         if context.launchTime.isActivePrewarm {
             // Set `active_pre_warm` attribute to true in case
-            // of pre-warmed app
+            // of pre-warmed app.
             attributes[Constants.activePrewarm] = true
-        } else {
+        } else if let launchTime = context.launchTime.launchTime {
             // Report Application Launch Time only if not pre-warmed
-            loadingTime = context.launchTime.launchTime.toInt64Nanoseconds
+            loadingTime = launchTime.toInt64Nanoseconds
+        } else if let launchDate = context.launchTime.launchDate {
+            // The launchTime can be `nil` if the application is not yet
+            // active (UIApplicationDidBecomeActiveNotification). That is
+            // the case when instrumenting a SwiftUI application that start
+            // a RUM view on `SwiftUI.View.onAppear`.
+            //
+            // In that case, we consider the time between the application
+            // launch and the view start as the application loadint time.
+            loadingTime = viewStartTime.timeIntervalSince(launchDate).toInt64Nanoseconds
```

Comment:
This logic was previously enabled here: https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/_Datadog_Private/ObjcAppLaunchHandler.m#L92

In a SwiftUI application, the first `SwiftUI.View/onAppear` is called before the `UIApplicationDidBecomeActiveNotification`, so we consider the `startView` as the end of the launch sequence. It is best to have this logic explicitly stated in RUM.